enhancements/windows-containers/csi-proxy.md

```diff
@@ -0,0 +1,189 @@
+tracking-link:
+  - https://issues.redhat.com/browse/OCPBU-465
+---
+
+# Enable running CSI drivers on Windows Nodes through CSI Proxy
+
+## Release Sign-off Checklist
+
+- [x] Enhancement is `implementable`
+- [x] Design details are appropriately documented from clear requirements
+- [x] Test plan is defined
+- [x] Operational readiness criteria is defined
+- [x] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+
+## Summary
+
+The goal of this enhancement is to enable the use of persistent storage for Windows workloads, without the use of
+in-tree storage drivers, [which are in the process of being deprecated and removed.](https://kubernetes.io/blog/2022/09/26/storage-in-tree-to-csi-migration-status-update-1.25/)
+
+This will be done by running [CSI Proxy](https://github.com/kubernetes-csi/csi-proxy) as a Windows service on each
+Windows node. csi-proxy is a pre-requisite for CSI node driver pods to interact with the underlying Windows storage
+APIs. With csi-proxy running on a cluster's Windows nodes, cluster administrators are free to deploy the CSI storage
+drivers of their choice as a Windows DaemonSet, enabling Windows pods on each node to have access to storage through
+PersistentVolumeClaims.
+
+## Motivation
+
+Windows nodes on OpenShift have had persistent storage available through the use of the in-tree storage drivers.
+Persistent storage is a necessary feature for production workloads on Kubernetes, and it is important that this
+functionality continue to work, even with the in-tree drivers being removed.
+
+### User Stories
+
+* As a developer using OpenShift, I want the Windows workloads I create to be able to read and write to storage which
+  persists when pods are removed and re-created.
+
+### Goals
+
+* Windows nodes configured by WMCO fulfill all the pre-requisites to run CSI node drivers.
+
+### Non-Goals
+
+* The Windows CSI storage drivers will not be distributed as part of OpenShift.
+  * At this time, we are not able to ship Windows containers images as all Windows container images are packaged
+with a Windows kernel. Red Hat has a policy to not ship 3rd party kernels for support reasons. In the future, we could
+use [host process container images](https://github.com/microsoft/windows-host-process-containers-base-image),
+however that would require CI and CPaaS pipelines to support building Windows containers.
+* Enabling the use of volumes which can be shared between Linux and Windows.
+
+## Proposal
+
+The [windows-machine-config-operator](https://github.com/openshift/windows-machine-config-operator/)(WMCO),
+will ensure that csi-proxy is running on the Windows nodes it configures. This will be done through the normal
+workflow for adding a service to the expected service list.
+
+The csi-proxy binary will be baked into the WMCO image. WMCO will copy the binary to each node at configuration
```

Comment:
Could be useful to call out how we will acquire the binary (built from source i assume)